pandas/plotting/_core.py

```diff
@@ -1980,50 +1980,160 @@ def plot_series(data, kind='line', ax=None,                    # Series unique
-    return_type : {None, 'axes', 'dict', 'both'}, default None
-        The kind of object to return. The default is ``axes``
-        'axes' returns the matplotlib axes the boxplot is drawn on;
-        'dict' returns a dictionary whose values are the matplotlib
-        Lines of the boxplot;
-        'both' returns a namedtuple with the axes and dict.
-
-        When grouping with ``by``, a Series mapping columns to ``return_type``
-        is returned, unless ``return_type`` is None, in which case a NumPy
-        array of axes is returned with the same shape as ``layout``.
-        See the prose documentation for more.
-
-    `**kwds` : Keyword Arguments
+        The size of the figure to create in matplotlib.
+    layout : tuple (rows, columns), optional
+        For example, (3, 5) will display the subplots
+        using 3 columns and 5 rows, starting from the top-left.
+    return_type : {None, 'axes', 'dict', 'both'}, default 'axes'
+        The kind of object to return. The default is ``axes``.
+
+        * 'axes' returns the matplotlib axes the boxplot is drawn on.
+        * 'dict' returns a dictionary whose values are the matplotlib
+          Lines of the boxplot.
+        * 'both' returns a namedtuple with the axes and dict.
+        * when grouping with ``by``, a Series mapping columns to
+          ``return_type`` is returned.
+
+          If ``return_type`` is `None`, a NumPy array
+          of axes with the same shape as ``layout`` is returned.
+    **kwds : Keyword Arguments, optional
         All other plotting keyword arguments to be passed to
-        matplotlib's boxplot function
+        :func:`matplotlib.pyplot.boxplot`.
 
     Returns
     -------
-    lines : dict
-    ax : matplotlib Axes
-    (ax, lines): namedtuple
+    result:
+        Options:
+
+        * ax : object of class
+          matplotlib.axes.Axes (for ``return_type='axes'``)
+        * lines : dict (for ``return_type='dict'``)
+        * (ax, lines): namedtuple (for ``return_type='both'``)
+        * :class:`~pandas.Series` (for ``return_type != None``
+          and data grouped with ``by``)
+        * :class:`~numpy.array` (for ``return_type=None``
+          and data grouped with ``by``)
+
+    See Also
+    --------
+    matplotlib.pyplot.boxplot : Make a box and whisker plot.
+    matplotlib.pyplot.hist : Make a histogram.
 
     Notes
     -----
     Use ``return_type='dict'`` when you want to tweak the appearance
     of the lines after plotting. In this case a dict containing the Lines
     making up the boxes, caps, fliers, medians, and whiskers is returned.
+
+    Examples
+    --------
+
+    Boxplots can be created for every column in the dataframe
+    by ``df.boxplot()`` or indicating the columns to be used:
+
+    .. plot::
+        :context: close-figs
+
+        >>> np.random.seed(1234)
+        >>> df = pd.DataFrame(np.random.randn(10,4),
+        ...                   columns=['Col1', 'Col2', 'Col3', 'Col4'])
```

Comment:
Is it a problem with the `.. plot::` directive to create `df` in a previous block? If it's not, in most documentation pages what I've seen is:
- `>>> df = ...`
- `>>> df` or `>>> df.head()` if it's long
- Some explanation of the function, `boxplot` in this case
- `df.boxplot()`

I think it's a bit more clear that the user first sees the data for the example, and then the explanations and the function usage is shown.

If that's a problem with `.. plot::` just leave it as it is, is not a big deal.

Reply:
It's been discussed before. I think that it is clearer for the user to know what dataframe is using since many types of grouping can be done and it is more explicit this way. For consistency,  I took the examples from Pandas Documentation: [Plotting with matplotlib: Box-plotting](http://pandas.pydata.org/pandas-docs/version/0.13.1/visualization.html#box-plotting) where a new dataframe is created for each example.

Reply:
Fair enough, I missed that discussion, sorry.